### ISSUE TITLE
Support Aggregations

### ISSUE THREAD

**Issue by PhaedrusTheGreek** [logstash-plugins/logstash-input-elasticsearch#58]
Not sure if this would be better supported by a totally different plugin, but I think it would make sense to support aggregation results.  

As an example, say we wanted to use a sum aggregation, by terms.

```
GET myserverlogs*/_search
{

   "size": 0,
   "aggs": {
      "types": {
         "terms": {
            "field": "hostname"
         },
         "aggs": {
            "connections": {
               "sum": {
                  "field": "connections"
               }
            }
         }
      }
   }
}
```

Results:
```
  "aggregations": {
      "types": {
         "doc_count_error_upper_bound": 0,
         "sum_other_doc_count": 0,
         "buckets": [
            {
               "key": "hostA",
               "doc_count": 90310,
               "total_connections": {
                  "value": 2344
               }
            },
            {
               "key": "hostB",
               "doc_count": 485,
               "total_connections": {
                  "value": 233
               }
            },
            {
               "key": "hostC",
               "doc_count": 485,
               "total_connections": {
                  "value": 123
               }
            }
         ]
      }
   }
```

Then in Logstash we could say something like:

```
input {
  elasticsearch {
    ...
    results_mode => "aggregation"
    source_array => "types.buckets"
  }
}
```

This could even be used to create aggregation based summary indices on a regular interval.  I remember @polyfractal was working on an ES plugin that did something similar last year.

**Comment by mlehotsky** [comment]
+1

**Comment by gitcdc** [comment]
+1

**Comment by shrv** [comment]
+1

**Comment by PhaedrusTheGreek** [comment]
Related news - ES 6.3 ships with a [Rollup API](https://www.elastic.co/guide/en/elasticsearch/reference/6.3/rollup-apis.html)

**Comment by MatonAnthony** [comment]
Is there any news on this? Is there a blocker for this? Would someone be willing to eventually mentor a Pull Request?

**Comment by francisca-lima** [comment]
Any solution?

**Comment by thekofimensah** [comment]
This would be great because I need to aggregate my massive dataset by a date histogram to compare one month against another month by event_id in logstash. I'm very surprized that this isnt supported ++++1

**Comment by pselvini** [comment]
+1

**Comment by genpepe** [comment]
+1

**Comment by tornis** [comment]
+1

**Comment by axelv** [comment]
+1

**Comment by ruif95** [comment]
+1

**Comment by thgruiz** [comment]
+1

**Comment by dneville-pep** [comment]
I'm honestly surprised that this isn't supported. + 1

**Comment by nnikushkin** [comment]
Hello guys! 

Could you please clarify whether if ETA of this feature release is known?

It will be really useful for me

Thank you!

**Comment by jsvd** [comment]
This feature has been released with [4.20.0](https://github.com/logstash-plugins/logstash-input-elasticsearch/blob/v4.20.0/CHANGELOG.md#4200) of this plugin. It should be included in the future Logstash 8.14.0 release (no target release date at this point).
